src/value/de.rs

```diff
@@ -41,22 +41,19 @@ impl<'de> Deserialize<'de> for Value {
             fn visit_i64<E>(self, i: i64) -> Result<Value, E>
                 where E: SError,
             {
-                Ok(Value::I64(i))
+                Ok(Value::Number(Number::from(i)))
             }
 
             fn visit_u64<E>(self, u: u64) -> Result<Value, E>
                 where E: SError,
             {
-                match NumCast::from(u) {
-                    Some(i) => Ok(Value::I64(i)),
-                    None => Ok(Value::String(u.to_string())),
-                }
+                Ok(Value::Number(Number::from(u)))
             }
 
             fn visit_f64<E>(self, f: f64) -> Result<Value, E>
                 where E: SError,
             {
-                Ok(Value::F64(f))
+                Ok(Value::Number(Number::from_f64(f).expect("wrapping an existing float")))
```

Comment:
No requirement really. I'll go your suggested route. I'm mostly just following what the JSON team did.